Tests/StarknetTests/Utils/DevnetClient.swift

```diff
@@ -0,0 +1,319 @@
+import Foundation
+
+import Starknet
+
+protocol DevnetClientProtocol {
+    var gatewayUrl: String { get }
+    var feederGatewayUrl: String { get }
+    var rpcUrl: String { get }
+    
+    func start()
+    func close()
+    
+    func prefundAccount(address: Felt)
+    func deployAccount(name: String) -> DeployAccountResult
+    func deployContract(contractPath: String) -> TransactionResult
+    func declareContract(contractPath: String) -> TransactionResult
+    func readAccountDetails(accountName: String) -> AccountDetails
+}
+
+struct AccountDetails: Codable{
+    var privateKey: Felt
+    var publicKey: Felt
+    var address: Felt
+    var salt: Felt
+    
+    enum CodingKeys: String, CodingKey {
+        case privateKey = "private_key"
+        case publicKey = "public_key"
+        case address
+        case salt
+    }
+    
+    init(privateKey: Felt, publicKey: Felt, address: Felt, salt: Felt) {
+        self.privateKey = privateKey
+        self.publicKey = publicKey
+        self.address = address
+        self.salt = salt
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.privateKey = try container.decode(Felt.self, forKey: .privateKey)
+        self.publicKey = try container.decode(Felt.self, forKey: .publicKey)
+        self.address = try container.decode(Felt.self, forKey: .address)
+        self.salt = try container.decode(Felt.self, forKey: .salt)
+        //try verifyTransactionIdentifiers(container: container, codingKeysType: CodingKeys.self)
+    }
+}
+
+struct TransactionResult {
+    var address: Felt
+    var hash: Felt
+}
+
+struct DeployAccountResult{
+    var details: AccountDetails
+    var txHash: Felt
+}
+
+enum DevnetClientError: Error {
+    case invalidTestPlatform
+}
+
+// Due to DevnetClient being albe to run only on a macos, this
+// factory method will throw, when ran on any other platform.
+func makeDevnetClient() throws -> DevnetClientProtocol {
+#if os(macOS)
+    return DevnetClient()
+#endif
+    
+    throw DevnetClientError.invalidTestPlatform
+}
+
+#if os(macOS)
+
+class DevnetClient: DevnetClientProtocol {
+    private let host: String
+    private let port: Int
+    private let seed: Int
+    // Paths still arent working properly, for now I'm using absolute paths
+    private let accountDirectory = "/Users/jakub/test"
+    private let baseUrl: String
+   
+    private var isDevnetRunning = false
+    private var devnetProcess: Process!
+    
+    let gatewayUrl: String
+    let feederGatewayUrl: String
+    let rpcUrl: String
+    
+    init(_host: String = "0.0.0.0", _port: Int = 5050, _seed: Int = 1053545547) {
+        host = _host
+        port = _port
+        seed = _seed
+        baseUrl = "http://\(host):\(port)"
+        gatewayUrl = "\(baseUrl)/gateway"
+        feederGatewayUrl = "\(baseUrl)/feeder_gateway"
+        rpcUrl = "\(baseUrl)/rpc"
+    }
```

Comment:
The common style in swift is to name init arguments the same way, and use e.g. `self.host = host`